### ISSUE TITLE
How much should be done training?

### ISSUE THREAD

**Issue by danchokobo** [david-gpu/deep-makeover#8]
During the making training by use of  => python3 dm_main.py --run train
it is generated only three bathes, but result not so pretty good. So I started training again, but results are similar to previous. @david-gpu How much time I should run this command? help me

**Comment by danchokobo** [comment]
Train time is 120, by that time training mode processed only three picture🙄

**Comment by subzerofun** [comment]
Read my reply here: https://github.com/david-gpu/deep-makeover/issues/7

You have to check first if your Mac even has a suitable graphics card. Your Theano has probably used your CPU – and if you have a MacBook the processor is too slow for this kind of computation.

I have tried my best to explain it in the link above. You should look at the Readme more carefully, David wrote that he has used a Nvidia GTX 1080 for training.

**Comment by danchokobo** [comment]
So sad story:/ Yeap my Mac is slow, because it is not pro just MacBook air2013:/ I will try it from another mac:(

**Comment by danchokobo** [comment]
@subzerofun if my Mac does nvidia, I can just replace it by installation CUDA?

**Comment by danchokobo** [comment]
Sorry if it does not have nvidia? @subzerofun

**Comment by subzerofun** [comment]
Sorry, but the Air models do not have a dedicated (extra) graphics card. The graphics processing is integrated in the CPU. And it doesn't work with CUDA, that's only for NVidia cards. The equivalent from AMD is OpenCL.

The processors of the MacBook Air models are intentionally "slow" (low clock speed) – they are build to consume little energy.

There are other options, like renting an Amazon server with CUDA support. But they're not that easy to set up (for a beginner) – you would need experience with Linux. And they are not free of course.

A laptop is not really an ideal machine for training neural nets. There are a few exceptions, like machines for 3d work or gaming laptops with high-end GPUs, but that's of no use to you.

If you are only interested in trying out the project i could upload my finished training weights. I don't think that the image generation would take that long, even on your MacBook. I have no idea if that would work, but one can only try.

**Comment by danchokobo** [comment]
@subzerofun thank u so much. It will be great if you send your trained project. Also I will try to upload project to the amazon server.

**Comment by danchokobo** [comment]
@subzerofun @david-gpu   Hi! OKAY I will try do it on more better computer with NVIDIA cards, I just want to know,  what I can do after completing all trainings? How I can test it by real images of my friends? Should I put into dataset those images of my friends? then again I should run training mode? or type something different into command line? Help me:)

**Comment by danchokobo** [comment]
@subzerofun also will be helpful your finished training weights for my MacBook 👍

**Comment by subzerofun** [comment]
@danchokobo 

> Should I put into dataset those images of my friends?

No, training should only be run with the aligned celeba dataset images. And you only need to run the training once for one mode (see below) until you get good results in the output pngs. There are four modes for training: 

```python
('`train_mode` must be one of: `ftm`, `mtf`, `ftf` or `mtm`')
```
"ftm" (female to male) is when your input image is **f**emale and output will be **m**ale
"mtf" = male to female
"ftf" = female to female
"mtm" = male to male

I think this would be the cmd line for a specific train mode:
`python3 dm_main.py --run train --train_mode "THREE-LETTER-TRAIN-MODE-WITHOUT-QUOTES" --train_time "TRAIN-TIME-IN-MINUTES"`

You could train four models, for all different modes. But you would have to change the name of (or move it to a different directory) of your `checkpoint` folder before starting a different training mode or else the last one will be overwritten.

> How I can test it by real images of my friends?
> or type something different into command line?

The code is:

`python3 dm_main.py --run inference --infile "YOUR-INPUT-IMAGE-WITHOUT-QUOTES" --outfile "OUTPUT-FILENAME.png"
`

But remember it will only use the gender conversion mode from your last training, training data gets loaded from the `checkpoint` folder.

> also will be helpful your finished training weights for my MacBook

I've accidentally overwritten my last training model, i will need to start a new one. I will write you when it's finished. Will probably take 1-2 days because my GPU is busy right now with processing another project.
